auth/kerberos/src/krb.cpp

```diff
@@ -1129,7 +1135,9 @@ std::string retrieve_secret_from_ecs_config(std::string ecs_variable_name)
         results = split_string(line, '=');
         std::string key = results[0];
         std::string value = results[1];
-        if ( ecs_variable_name.compare( key ) == 0 )
+        if ( !contains_invalid_characters_in_credentials(value) && ecs_variable_name.compare( key
```

Comment:
nit: Would this be better in one line? or have the whole second clause on its own line?

Reply:
addressed the comment